src/mono/CMakeLists.txt

```
@@ -231,6 +231,16 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
   # sys/random.h exists, but its not found
   set(HAVE_SYS_RANDOM_H 1)
   set(INTERNAL_ZLIB 1)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "WASI")
+  set(HOST_WASI 1)
+  add_definitions(-D_WASI_EMULATED_SIGNAL -D_WASI_EMULATED_MMAN)
+  add_definitions(-DNO_GLOBALIZATION_SHIM)
+  add_definitions(-D_THREAD_SAFE)
+  # FIXME: Is there a cmake option for this ?
+  set(DISABLE_SHARED_LIBS 1)
```

From https://cmake.org/cmake/help/latest/guide/tutorial/Selecting%20Static%20or%20Shared%20Libraries.html
```suggestion
  option(BUILD_SHARED_LIBS "Disabling shared libraries for WASI" OFF)
```